File: backend/models/vehicle.py
from sqlalchemy import Column, String, Integer, DateTime, Float
from sqlalchemy.orm import relationship
from datetime import datetime
from uuid import uuid4

from backend.base import Base


class Vehicle(Base):
    __tablename__ = "vehicles"

    id = Column(String, primary_key=True, default=lambda: str(uuid4()))
    user_id = Column(String, nullable=False)

    name = Column(String, nullable=False)
    model = Column(String, nullable=False)
    year = Column(Integer, nullable=False)

    registration_number = Column(String, nullable=False)

    mileage = Column(Integer, default=0)
    fuel_level = Column(Integer, default=100)

    created_at = Column(DateTime, default=datetime.utcnow)

    # 🤖 AI fields
    ai_risk_level = Column(String, nullable=True)   # LOW / MEDIUM / HIGH
    ai_failure_probability = Column(Float, nullable=True)
    ai_component = Column(String, nullable=True)
    ai_last_analyzed = Column(DateTime, nullable=True)

    # 🔗 Relationship
    appointments = relationship(
        "Appointment",
        back_populates="vehicle",
        cascade="all, delete-orphan"
    )

File: proactive-frontend/my-project/src/lib/normalizers/vehicle.ts
export function normalizeVehicle(v: any) {
  return {
    id: v.id,
    userId: v.user_id,
    name: v.name,
    model: v.model,
    year: v.year,
    registrationNumber: v.registration_number,
    createdAt: v.created_at,
  };
}
export function normalizeDashboardVehicle(v: any) {
  const health = Math.max(
    0,
    Math.round(100 - (v.ai_failure_probability ?? 0) * 100)
  );

  let status: "healthy" | "warning" | "critical" = "healthy";
  if (v.ai_risk_level === "MEDIUM") status = "warning";
  if (v.ai_risk_level === "HIGH") status = "critical";

  return {
    id: v.id,
    name: v.name,
    registration: v.registration_number,
    mileage: v.mileage,
    fuel: v.fuel_level,

    // 🔥 AI-derived fields
    health,
    status,
    riskLevel: v.ai_risk_level,
    failureProbability: v.ai_failure_probability,
    affectedComponent: v.ai_component,
    confidence: v.ai_confidence,
    lastAnalyzed: v.ai_last_analyzed,

    nextService: v.next_service_date ?? null
  };
}
